### PR TITLE
Extra message for New Garage

### PR DESCRIPTION
**Pull request by igorkis-scrts**
## What type of PR is this.
1. [ ] Bug
2. [ ] Change
3. [x] Enhancement

### What have you changed and why?
Information:
    Old, pre-2.5.0-garage had the ability to show some extra messages during vehicle placement process. This PR implements same thing for Hakon's Garage too.
This might be useful in some non-general use cases: some additional hints, guidances or simply vehicle price can be shown to player.

image

 
### Please specify which Issue this PR Resolves.
-

### Please verify the following and ensure all checks are completed.
-

1. [x] Have you loaded the mission in LAN host?
2. [x] Have you loaded the mission on a dedicated server?

### Is further testing or are further changes required?
1. [x] No
2. [ ] Yes (Please provide further detail below.)

### How can the changes be tested?
For example, fn_addFIAVeh.sqf:
```sqf
private _extraMessage = format ["Buying vehicle for $%1.", vehiclePurchase_cost];

[_typeVehX, "BUYFIA", [], nil, nil, nil, false, _extraMessage] call HR_GRG_fnc_confirmPlacement;
```

### CODE REVIEW

File: A3A/addons/Garage/Core/fn_confirmPlacement.sqf [hunk 1]
```diff
@@ -34,12 +35,12 @@ params [
...
     , "_state"
     , ["_useGRGPool", false, [false]]
+    , ["_extraText", "", [""]]
 ];
 
 if (!isClass (configFile >> "CfgVehicles" >> _class)) exitWith {HR_GRG_placing = false};
 if (isNil "HR_GRG_curTexture") then {HR_GRG_curTexture = []};
 if (isNil "HR_GRG_curAnims") then {HR_GRG_curAnims = []};
-if (!isNil "HR_GRG_placing" && {HR_GRG_placing}) exitWith {Error_1("already placing, params: %1", _this)};
 HR_GRG_placing = true;
```
Comment (HakonRydland) [inline A3A/addons/Garage/Core/fn_confirmPlacement.sqf:44]: this blocker check is required, add it back in

File: A3A/addons/Garage/Core/fn_confirmPlacement.sqf [hunk 2]
```diff
@@ -383,7 +385,12 @@ HR_GRG_EH_EF = addMissionEventHandler ["EachFrame", {
 
     //render keybind hint
     private _text = switch HR_GRG_validPlacement do {
-        case 0: {localize "STR_HR_GRG_Feedback_CP_Rotation"};
+        case 0: {
+            [
+                localize "STR_HR_GRG_Feedback_CP_Rotation",
+                format ["%1%2", HR_GRG_CP_extraText, localize "STR_HR_GRG_Feedback_CP_Rotation"]
```
Comment (HakonRydland) [inline A3A/addons/Garage/Core/fn_confirmPlacement.sqf:391]: add the linebreak here, it should not be the responsibility of the user to add it

File: A3A/addons/core/functions/REINF/fn_addFIAveh.sqf [hunk 3]
```diff
@@ -27,6 +27,6 @@ if (_resourcesFIA < vehiclePurchase_cost) exitWith {["Add Vehicle", format ["You
 vehiclePurchase_nearestMarker = [markersX select {sidesX getVariable [_x,sideUnknown] == teamPlayer},player] call BIS_fnc_nearestPosition;
 if !(player inArea vehiclePurchase_nearestMarker) exitWith {["Add Vehicle", "You need to be close to the flag to be able to purchase a vehicle."] call A3A_fnc_customHint;};
 
-private _extraMessage =	format ["Buying vehicle for $%1.", vehiclePurchase_cost];
+private _extraMessage =	format ["Buying vehicle for $%1.<br/>", vehiclePurchase_cost];
```
Comment (HakonRydland) [inline A3A/addons/core/functions/REINF/fn_addFIAveh.sqf:30]: reminder to remove the linebrak from here

**Review by HakonRydland — CHANGES_REQUESTED** [review]

File: A3A/addons/Garage/Core/fn_confirmPlacement.sqf (hunk 1 above)
Comment (igorkis-scrts) [reply A3A/addons/Garage/Core/fn_confirmPlacement.sqf]: Whoops
Fixed in [commit](901c137ff212c6a66c62ebbf6ad2bb17803c03b0)

File: A3A/addons/Garage/Core/fn_confirmPlacement.sqf (hunk 2 above)
Comment (igorkis-scrts) [reply A3A/addons/Garage/Core/fn_confirmPlacement.sqf]: Fixed in [commit](901c137ff212c6a66c62ebbf6ad2bb17803c03b0)

File: A3A/addons/core/functions/REINF/fn_addFIAveh.sqf (hunk 3 above)
Comment (igorkis-scrts) [reply A3A/addons/core/functions/REINF/fn_addFIAveh.sqf]: Fixed in [commit](901c137ff212c6a66c62ebbf6ad2bb17803c03b0)

**Review by HakonRydland — APPROVED** [review]

**Review by Bob-Murphy — APPROVED** [review]
lgtm

**Comment by igorkis-scrts** [comment]
It's a little bit late as PR already merged, but - make sure that the extra message's structured text has at least the same size as the rest.
I've noticed today that font of extra message is bigger than the rest of the hint, so it needs to be properly formatted. This could be done either in message or automatically applied in string concatenation. 

For example:
1.
```xml
<Original>&lt;t size='0.6'&gt;Select vehicle position.&lt;br/&gt;Price: %1%2&lt;/t&gt;</Original>
```
2. 
```sqf
format ["<t size='0.6'>%</t><br/>%2", HR_GRG_CP_extraText, localize "STR_HR_GRG_Feedback_CP_Rotation"]
```

**Pull request merged**